### ISSUE TITLE
Could not load type 'Microsoft.AspNetCore.Mvc.Internal.ITypeActivatorCache' from assembly 'Microsoft.AspNetCore.Mvc.Core, Version=3.0.0.0

### ISSUE THREAD

**Issue by mscrivo** [simpleinjector/SimpleInjector#748]
**Describe the bug**
Testing out upgrading a currently working asp.net core 2.2 app with SimpleInjector integration to 3.0-preview9. The app won't start because of an issue locating an internal asp.net core type: `ITypeActivatorCache`

**Expected behavior**
App doesn't throw an exception.

**Actual behavior**
Here's the exception:

```
   at SimpleInjector.SimpleInjectorAspNetCoreBuilderMvcExtensions.<>c__DisplayClass1_0.<AddTagHelperActivation>b__1(IServiceProvider p)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitCache(ServiceCallSite callSite, RuntimeResolverContext context, ServiceProviderEngineScope serviceProviderEngine, RuntimeResolverLock lockType)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite singletonCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass1_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.AspNetCore.Mvc.Razor.RazorPageBase.get_TagHelperFactory()
   at Microsoft.AspNetCore.Mvc.Razor.RazorPageBase.CreateTagHelper[TTagHelper]()
   at AspNetCore.Views_Shared_Redirect.<ExecuteAsync>d__9.MoveNext()
```

**To Reproduce**
Create .net core 3 mvc app with SI integration and try running it.

**Additional context**
.NET Core 3.0 preview 9
SimpleInjector 4.7.1

**Comment by mscrivo** [comment]
Note, this can be bypassed for now by commenting out:

`.AddTagHelperActivation();`

**Comment by dotnetjunkie** [comment]
## Steps to reproduce:

* Download Visual Studio (2019) 16.3 preview 4 from https://visualstudio.microsoft.com/vs/preview/
* Install VS with the "Web Application" Workload
* Start the preview and create a new project type "ASP.NET Core Web Application" for "ASP.NET Core 3.0" with the "Web Application (Model-View-Controller)" template.
* Add the "SimpleInjector.Integration.AspNetCore.Mvc" Version="4.7.1" NuGet package to the WebApplication project.
* Add the Simple Injector bootstrapping to the `Startup.cs`, making it look as follows:

``` c#
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;
using SimpleInjector;

namespace WebApplication1
{
    public class Startup
    {
        private readonly Container container = new Container();

        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllersWithViews();

            services.AddLogging();
            services.AddLocalization(options => options.ResourcesPath = "Resources");

            services.AddSimpleInjector(container, options =>
            {
                options.AddAspNetCore()
                    .AddControllerActivation()
                    .AddViewComponentActivation()
                    .AddPageModelActivation()
                    .AddTagHelperActivation();
            });
        }

        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            app.UseSimpleInjector(container, options =>
            {
                options.UseLogging();
                options.UseLocalization();
            });

            InitializeContainer();

            container.Verify();

            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                app.UseExceptionHandler("/Home/Error");
            }
            app.UseStaticFiles();

            app.UseRouting();

            app.UseAuthorization();

            app.UseEndpoints(endpoints =>
            {
                endpoints.MapControllerRoute(
                    name: "default",
                    pattern: "{controller=Home}/{action=Index}/{id?}");
            });
        }

        private void InitializeContainer()
        {
        }
    }
}
```

* Run the application.

**Comment by dotnetjunkie** [comment]
[bug-748 branch](https://github.com/simpleinjector/SimpleInjector/tree/bug-748) created.

**Comment by dotnetjunkie** [comment]
I just released patch release [v4.7.2](https://github.com/simpleinjector/SimpleInjector/releases/tag/v4.7.2). That release fixes this issue.

**Comment by mscrivo** [comment]
Thanks Steven!
